lang/syn/src/idl/file.rs

```diff
@@ -191,6 +255,59 @@ pub fn parse(
     }))
 }
 
+/// Dependency of a Rust crate (an another crate).
+struct CargoDependency {
+    path: PathBuf,
+    features: Option<Vec<String>>,
+}
+
+/// Parse the Cargo metadata and return a map of dependencies, their source
+/// paths (inside Cargo cache), and their enabled features from Cargo metadata
+/// of the main crate.
+fn dependencies<P>(cargo_path: P) -> Result<HashMap<String, CargoDependency>>
+where
+    P: AsRef<Path>,
+{
+    let config = cargo::Config::default()?;
+    let ws = Workspace::new(cargo_path.as_ref(), &config)?;
+
+    let dependency_features = ws
+        .members()
+        .flat_map(|member| {
+            member.dependencies().iter().map(move |dependency| {
+                let features: Vec<String> = dependency
+                    .features()
+                    .iter()
+                    .map(|feature| feature.to_string())
+                    .collect();
+                (dependency.package_name().to_string(), features)
+            })
+        })
+        .collect::<HashMap<String, Vec<String>>>();
+
+    // TODO(vadorovsky): Try to do it just with `cargo` crate.
```

Comment:
Did you do this with your dummy crate, or is this todo for sth else?

Reply:
I used https://crates.io/crates/cargo_metadata to parse information about features of dependencies. Which is kinda crappy, because it's using `Command` to call cargo - https://github.com/oli-obk/cargo_metadata/blob/main/src/lib.rs#L737-L744

I was trying to avoid that and use the internals of [cargo's source code](https://crates.io/crates/cargo), but I failed, I didn't find any public function or method which exposes metadata of dependencies yet. I'm sure there is something I missed, but I just didn't get time to dig into cargo's code that much.

I will update the comment.